src/statistics.rs

```diff
@@ -0,0 +1,53 @@
+//! # Thermostat_EEM temperature telemetry processing
+
+use serde::Serialize;
+
+/// Statistics telemetry struct.
+#[derive(Serialize, Copy, Clone, Default, Debug)]
+pub struct Statistics {
+    mean: f32,
+    min: f32,
+    max: f32,
+}
+
+impl From<Buffer> for Statistics {
+    /// Process temperature buffer. Calculates mean, returns the finalized Statistics type
```

Comment:
`From` should not modify the input. Either it consumes it (then modifying is pointless) or it gets it by immutable reference (then it can't).

Reply:
I actually committed the same error twice today, earlier with `mut self` instead of `&mut self`..
Easy to just forget.